nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -1101,11 +1279,15 @@ def forward(
             hidden_states = self.input_tensor
 
         if encoder_output is not None:
-            encoder_output = encoder_output.transpose(0, 1).contiguous()
+            if len(encoder_output.shape) == 5:
```

Comment:
Is it possible to do a stronger check here?

Reply:
Since the encoder_output is a tensor. The only difference between normal seq-to-seq encoder output and retrieval embedding is the order of dimension. One solution is to add this as an argument for `forward` method, but I guess it causes the problem of bloated arguments and need a few more if statements in the forward method.

Reply:
I added `retrieved_emb` as an argument.